ExRam.Gremlinq.Core/Models/GraphModel.cs

```diff
@@ -21,7 +22,103 @@ private enum GraphElementType
+            {
+                switch (expression)
+                {
+                    case MemberExpression leftMemberExpression:
+                        {
+                            return GetIdentifier(leftMemberExpression.Expression.Type, leftMemberExpression.Member.Name);
+                        }
+                    case ParameterExpression leftParameterExpression:
+                        {
+                            return GetIdentifier(leftParameterExpression.Type, leftParameterExpression.Name);
+                        }
+                    default:
+                        throw new ExpressionNotSupportedException(expression);
+                }
+            }
+
+            public virtual object GetIdentifier(Type elementType, string memberName)
+            {
+                if (string.Equals(memberName, "id", StringComparison.OrdinalIgnoreCase) || string.Equals(memberName, "label", StringComparison.OrdinalIgnoreCase))
+                {
+                    var graphElementType = ElementTypes
+                        .GetOrCreateValue(this)
+                        .GetOrAdd(
+                            elementType,
+                            closureElementType =>
+                            {
+                                if (elementType == typeof(IVertex) || VerticesModel.TryGetConstructiveLabel(elementType).IsSome)
+                                    return GraphElementType.Vertex;
+
+                                if (elementType == typeof(IEdge) || EdgesModel.TryGetConstructiveLabel(elementType).IsSome)
+                                    return GraphElementType.Edge;
+
+                                return typeof(IVertexProperty).IsAssignableFrom(elementType)
+                                    ? GraphElementType.VertexProperty
+                                    : GraphElementType.None;
+                            });
+
+                    if (graphElementType != GraphElementType.None)
+                    {
+                        if (string.Equals(memberName, "id", StringComparison.OrdinalIgnoreCase))
+                            return T.Id;
+
+                        if (string.Equals(memberName, "label", StringComparison.OrdinalIgnoreCase))
+                            return T.Label;
+                    }
+                }
+
+                return memberName;
+            }
+        }
+
+        private sealed class CamelcaseGraphModel : GraphModelBase
+        {
+            private sealed class CamelcaseGraphElementModel : IGraphElementModel
+            {
+                private readonly IGraphElementModel _baseModel;
+
+                public CamelcaseGraphElementModel(IGraphElementModel baseModel)
+                {
+                    _baseModel = baseModel;
+                }
+
+                public Option<string> TryGetConstructiveLabel(Type elementType) => _baseModel.TryGetConstructiveLabel(elementType).Map(x => x.ToCamelCase());
```

Comment:
I'd like to separate lowercase labels from lowercase identifiers, and so for CamelCase. So could we have a `WithCamelcaseIdentifiers` extension and a `WithCamelcaseLabels` extension, if you need both?

Reply:
or just the latter, if you just need labels in camel case.

Reply:
I broke out `WithCamelcase` into `WithCamelcaseLabels` and `WithCamelcaseIdentifiers`. I also added in some tests to confirm it works in either order of adding them to the model.